kano_profile/badges.py

```diff
@@ -390,6 +390,31 @@ def save_app_state_variable_with_dialog(app_name, variable, value):
     save_app_state_with_dialog(app_name, data)
 
 
+def update_upwards_with_dialog(app_name, variable, value):
```

Comment:
Yes, the only difference between the regular fns and the ones with the `_with_dialog` suffixes is that the latter ones calculate whether there has been a change in the unlocked badges and display a notification accordingly